### ISSUE TITLE
fs_group and run_as_user is being defaulted to 0 while using kubernetes executor and airflow 1.10.10

### ISSUE THREAD

**Issue by aneesh-joseph** [apache/airflow#8683]
**Apache Airflow version**: 1.10.10

**Kubernetes version (if you are using kubernetes)** (use `kubectl version`): Not version specific

**Environment**:

- **Cloud provider or hardware configuration**:
- **OS** (e.g. from /etc/os-release):
- **Kernel** (e.g. `uname -a`):
- **Install tools**:
- **Others**:

**What you expected to happen**:

If I use the `KubernetesExecutor` and do not set a `fs_group` in airflow.cfg I would expect that the the `fs_group` is not set in the securityContext while creating the pod..It is also expected that the `fs_group` is set to 65533 if git sync with ssh is being used and `fs_group` is not explicitly set..

**What happened**:

If `fs_group` is not set, it gets defaulted to 0 - https://github.com/apache/airflow/blob/1.10.10/airflow/contrib/executors/kubernetes_executor.py#L295-L300 . Since this is being defaulted to 0, the logic to set it  to 65533 
 doesn't seem to come into play https://github.com/apache/airflow/blob/1.10.10/airflow/contrib/kubernetes/worker_configuration.py#L238-L239

The code comments suggest that it should return blank if  not set, but it returns 0 instead

```
    # pod security context items should return integers
    # and only return a blank string if contexts are not set.
    def _get_security_context_val(self, scontext):
        val = conf.get(self.kubernetes_section, scontext)
        if not val:
            return 0
        else:
            return int(val)
```
On the master branch, this is defined as expected

**How to reproduce it**:

Remove `fs_group` config from airflow.cfg and examine the worker pods created by KubernetesExecutor using  `kubectl get pod <worker_pod_name> -o` and examine the securityContext,

**Anything else we need to know**:

**Comment by aneesh-joseph** [comment]
this is already fixed on  master, can this be fixed on 1.10 branch as well please

**Comment by ashb** [comment]
@aneesh-joseph If you find us the change that made this on master we can make sure it makes it in to 1.10.11.

**Comment by aneesh-joseph** [comment]
@ashb , awesome, so  going through the github history, this problem crept in while sorting some pylint errors - https://github.com/apache/airflow/commit/47801057989046dfcf7b424ce54afee103803815 (master) and https://github.com/apache/airflow/commit/b037ba8f5037f338913dfa78b195fafd5a14e5dd (1.10) and was fixed on master with https://github.com/apache/airflow/commit/133085eb47e04683ce3dca52b967aa41f8139613 (PR  https://github.com/apache/airflow/pull/6312) on master . can we bring this change to 1.10.11

**Comment by ashb** [comment]
This has now been included in the v1-10-test branch as https://github.com/apache/airflow/commit/7ff7352a629a60106f6743257adad82367217c29 (commit SHA may yet change before release) -- closing this as it'll be fixed in 1.10.11
